### ISSUE TITLE
KPropertyPath div for nested document using @Id incorrect

### ISSUE THREAD

**Issue by hameno** [spring-projects/spring-data-mongodb#4330]
I'm having some issues with queries that use the following setup:

```kotlin
import org.springframework.data.annotation.Id
import org.springframework.data.mapping.div
import org.springframework.data.mapping.toDotPath
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "data")
data class Data(
    @Id
    val id: String,
    val nested: NestedData
)

data class NestedData(
    @Id
    val id: String
)

val path = (Data::nested / NestedData::id).toDotPath()
println(path) // prints 'nested.id' instead of 'nested._id'
```

Am I doing something wrong?

Using Spring Boot 2.7.9 / org.springframework.data:spring-data-mongodb:3.4.8

**Comment by christophstrobl** [comment]
This is the intended behaviour as the path is based on the actual properties and not their store representation. The mapping layer takes care of the field property <> name alignment. 
Using `@Id` on nested elements is not recommended. Id values should be assigned to top level domain types. You can store nested elements that have an id property as `id` and not `_id` via `@Field("id") val id : String`.
Is there an actual query that is failing?

**Comment by hameno** [comment]
Yes, my aggregations using lookup pipelines are failing. When I write the path manually it works.
Can't change the model now unfortunately, and there it's required as "_id"

**Comment by christophstrobl** [comment]
Can you share a complete [minimal sample](https://stackoverflow.com/help/minimal-reproducible-example) (something that we can unzip or git clone, build, and deploy) that reproduces the problem, so we can see what's going wrong when with `$lookup`?

**Comment by hameno** [comment]
I'll try when I have some time, unfortunately very busy right now 😬
